### PR TITLE
EN-41224: Add extract isoyear

### PR DESCRIPTION
**Pull request by QMalcolm**
Add SoQL function to extract the isoyear from a floating timestamp.

### CODE REVIEW

File: soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala [hunk 1]
```diff
@@ -453,6 +453,9 @@ object SoQLFunctions {
   val FloatingTimeStampExtractWoy = mf("floating timestamp extract week of year", FunctionName("date_extract_woy"), Seq(SoQLFloatingTimestamp), Seq.empty, SoQLNumber)(
```
Comment (znep) [inline soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala:453]: I might change the string to be something like `floating timestamp extract ISO week of year` and/or add a comment that `woy` is ISO week just to leave a few crumbs to make that more obvious to folks in the future despite the function not being named `iso`.

File: soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala [hunk 2]
```diff
@@ -453,6 +453,9 @@ object SoQLFunctions {
...
     NoDocs
   )
+  val FloatingTimestampExtractIsoY = mf("floating timestamp extract isoyear", FunctionName("date_extract_iso_y"), Seq(SoQLFloatingTimestamp), Seq.empty, SoQLNumber)(
+    NoDocs
```
Comment (rozap) [inline soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala:457]: `NoDocs` :(

If you write a doc, it'll show up in the new query editor :)

Comment (QMalcolm) [reply soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala:457]: Where should I write said doc? Is there an example?

Comment (QMalcolm) [reply soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala:463]: The discussion happened elsewhere. There is another branch which is doing a bunch of the initial doc write-ups (https://github.com/socrata-platform/soql-reference/tree/cduranti/add-docs). Once that branch is merged, I'll loop back around making a new pr to add the doc for this new function.

**Pull request merged**